packages/core/src/actors.ts

```diff
@@ -440,7 +331,7 @@ export function createMachineBehavior<
 
       if (event.type === stopSignalType) {
         service?.stop();
-        subscription && subscription.unsubscribe(); // TODO: might not be necessary
+        subscription?.unsubscribe(); // TODO: might not be necessary
```

Comment:
Ye, it probably isn't - although there is one scenario to be considered here. What about the state that is a result of the `service.stop()` call? Should it still be broadcasted to subscribers? Do we need to? 

I don't know how it works right now, nor I don't yet have an opinion on how this should behave.